src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtil.java

```diff
@@ -83,14 +83,7 @@ private CheckUtil() {
      * @return {@code FullIdent} for given type.
      */
     public static FullIdent createFullType(final DetailAST typeAST) {
-        DetailAST ast = typeAST;
-
-        // ignore array part of type
-        while (ast.findFirstToken(TokenTypes.ARRAY_DECLARATOR) != null) {
-            ast = ast.findFirstToken(TokenTypes.ARRAY_DECLARATOR);
-        }
-
-        return FullIdent.createFullIdent(ast.getFirstChild());
+        return FullIdent.createFullIdent(typeAST.getFirstChild());
```

Comment:
@romani What is your thoughts on removing this util method if all it does is call another method?

Reply:
FWIW, I left this since this is part of our API, and I did not want to remove a method that could be used in custom checks.

Reply:
If this method is a candidate for removal, then it should be marked as deprecated.

Reply:
let mark it as deprecated, nobody will notice this, as remove in few releases as breaking compatibility ...

Reply:
This PR is already heavy, I have opened an issue at https://github.com/checkstyle/checkstyle/issues/10049